src/containers/reservations/Accordion.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useState, useRef } from 'react';
+import { CaretDownOutlined, CaretUpOutlined } from '@ant-design/icons';
+import styled from 'styled-components';
+
+const AccordionButton = styled.button`
+  background-color: #61802e;
+  padding: 5px;
+  border: none;
+  outline: none;
+  transition: background-color 0.4s ease;
+`;
+
+const AccordionTextDiv = styled.div`
+  padding: 18px;
+`;
+
+const AccordionContentDiv = styled.div`
+  overflow: auto;
+  transition: height 0.4s ease;
+`;
+
+const AccordionSectionDiv = styled.div`
+  display: flex;
+  flex-direction: column;
+`;
+
+const Accordion: React.FC = ({ children }) => {
+  const [setActive, setActiveState] = useState(false);
+  const content = useRef(document.createElement('div'));
```

Comment:
why's this needed?

Reply:
I need to have access to the scrollHeight of the AccordionContentDiv, but I don't know how to do that in React/TS without a ref. (I tried using `this`, but that was giving me the error message "TS7041: The containing arrow function captures the global value of 'this'."